webapp/src/webusb.tsx

```diff
@@ -470,18 +470,67 @@ function theme() {
 export function renderUnpairDialog() {
     const boardName = getBoardName();
     const header = lf("How to unpair your {0}", boardName);
-    const jsx = <div><p>
-        {lf("You can unpair your {0} if the WebUSB download is malfunctioning. Click on the lock icon and uncheck your device.", boardName)}
-    </p>
-        <img className="ui image centered medium" src={"./static/webusb/unpair.gif"} alt={lf("A gif showing how to unpair the {0}", boardName)} />
+    const unpairImg = theme().browserUnpairImage;
+    const jsx = <div>
+        <p>
+            {lf("You can unpair your {0} if the WebUSB download is malfunctioning.", boardName)}
+        </p>
+        <p>
+            {lf("Click on the icon on the left side of the address bar and uncheck your device.")}
+        </p>
+        {unpairImg && <img
+            className="ui image centered medium"
+            src={unpairImg}
+            alt={lf("A gif showing how to unpair the {0}", boardName)}
+        />}
     </div>
 
-    // TODO: show usb forget here
     const helpUrl = pxt.appTarget.appTheme.usbDocs
         && (pxt.appTarget.appTheme.usbDocs + "/webusb#unpair");
     return { header, jsx, helpUrl };
 }
 
+export async function showDeviceForgottenDialog(confirmAsync: ConfirmAsync) {
+    const boardName = getBoardName();
+    const deviceForgottenImage = theme().usbDeviceForgottenImage;
+    const columns = deviceForgottenImage ? "two" : "one";
+
+    const jsxd = () => (
+        <div className={`ui ${columns} column grid padded download-dialog`}>
+            <div className="column">
+                <div className="ui">
+                    <div className="content">
+                        <div className="description">
+                            {lf("Your {0} has been disconnected.", boardName)}
+                            <br />
+                            <br />
+                            {lf("Unplug it from your computer and disconnect any battery to fully reset it.")}
```

Comment:
Should they just click the reset button instead of disconnecting the battery?

Reply:
No, this is pushing towards a power cycle, not a soft reset. Most probably won't read this dialog anyway, but if they only do a power cycle it e.g. would probably be in an inconsistent state on the device connection (daplink) side if they immediately try to pair again which may or may not work (depending on whether the device firmware handles it, e.g. not storing any bad addresses / state / etc in a way our connection logic can't reset)